packages/nextjs/components/scaffold-eth/Contract/ContractUI.tsx

```diff
@@ -1,82 +1,194 @@
-import { useReducer } from "react";
+import { useEffect, useReducer, useState } from "react";
+import router from "next/router";
 import { ContractReadMethods } from "./ContractReadMethods";
 import { ContractVariables } from "./ContractVariables";
 import { ContractWriteMethods } from "./ContractWriteMethods";
 import { Abi } from "viem";
-import { Address, Balance } from "~~/components/scaffold-eth";
+import { HeartIcon } from "@heroicons/react/24/outline";
+import { BuidlGuidlLogo } from "~~/components/assets/BuidlGuidlLogo";
+import { Address, Balance, MethodSelector } from "~~/components/scaffold-eth";
 import { useNetworkColor } from "~~/hooks/scaffold-eth";
 import { useAbiNinjaState } from "~~/services/store/store";
 import { getTargetNetworks } from "~~/utils/scaffold-eth";
 
+type ExtractAbiFunction<T> = T extends { type: "function" } ? T : never; // @todo duplicate in MethodSelector
+type AbiFunction = ExtractAbiFunction<Abi[number]>;
```

Comment:
We can remove this and instead use `AbiFuntion` type from `abitype` : 

`import { AbiFunction} from "abitype";`

Although we don't have it as our direct dependency its still present due to viem and wagmi, we are using types from `abitype` at other places checkout [`contract.ts`](https://github.com/BuidlGuidl/abi.ninja/blob/556b0674d3bec775a532a2a66899eb97108a894b/packages/nextjs/utils/scaffold-eth/contract.ts#L8)

Reply:
Thanks! You are amazing!!!